p2p/src/disconnection_reason.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2021-2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::Display;
+
+use common::{chain::config::MagicBytes, primitives::time::Time};
+
+use p2p_types::services::Services;
+
+use crate::{
+    error::{ConnectionValidationError, P2pError},
+    protocol::MIN_SUPPORTED_PROTOCOL_VERSION,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum DisconnectionReason {
+    AddressBanned,
+    AddressDiscouraged,
+    PeerEvicted,
+    FeelerConnection,
+    ConnectionFromSelf,
+    ManualDisconnect,
+    PingIgnored,
+    SyncRequestsIgnored,
+    TooManyInboundPeersAndThisOneIsDiscouraged,
+    TooManyInboundPeersAndCannotEvictAnyone,
+    UnsupportedProtocol,
+    TimeDiff {
+        remote_time: Time,
+        accepted_peer_time: std::ops::RangeInclusive<Time>,
+    },
+    DifferentNetwork {
+        our_network: MagicBytes,
+    },
+    NoCommonServices,
+    InsufficientServices {
+        needed_services: Services,
+    },
+}
+
+impl Display for DisconnectionReason {
```

Comment:
Is in it better to make DisconnectionReason an error using thiserror?

Reply:
I'm fine either way... I was just surprised that this is too verbose for what it's.